google/resource_compute_project_metadata_item.go

```diff
@@ -0,0 +1,178 @@
+
+	err = updateComputeCommonInstanceMetadata(config, projectID, key, &val)
+	if err != nil {
+		return err
+	}
+
+	d.SetId(key)
+
+	return nil
+}
+
+func resourceComputeProjectMetadataItemRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	projectID, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("[DEBUG] Loading project metadata: %s", projectID)
+	project, err := config.clientCompute.Projects.Get(projectID).Do()
+	if err != nil {
+		return fmt.Errorf("Error loading project '%s': %s", projectID, err)
+	}
+
+	md := flattenComputeMetadata(project.CommonInstanceMetadata.Items)
+	val, ok := md[d.Id()]
+	if !ok {
+		// Resource no longer exists
+		d.SetId("")
+		return nil
+	}
+
+	d.Set("key", d.Id())
+	d.Set("value", val)
+
+	return nil
+}
+
+func resourceComputeProjectMetadataItemUpdate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	projectID, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	if d.HasChange("value") {
+		key := d.Get("key").(string)
+		_, n := d.GetChange("value")
+		new := n.(string)
+
+		err = updateComputeCommonInstanceMetadata(config, projectID, key, &new)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func resourceComputeProjectMetadataItemDelete(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	projectID, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	key := d.Get("key").(string)
+
+	err = updateComputeCommonInstanceMetadata(config, projectID, key, nil)
+	if err != nil {
+		return err
+	}
+
+	d.SetId("")
+	return nil
+}
+
+func updateComputeCommonInstanceMetadata(config *Config, projectID string, key string, afterVal *string) error {
+	updateMD := func() error {
+		log.Printf("[DEBUG] Loading project metadata: %s", projectID)
+		project, err := config.clientCompute.Projects.Get(projectID).Do()
+		if err != nil {
+			return fmt.Errorf("Error loading project '%s': %s", projectID, err)
+		}
+
+		md := flattenComputeMetadata(project.CommonInstanceMetadata.Items)
+
+		val, ok := md[key]
+
+		if !ok {
+			if afterVal == nil {
+				// Asked to set no value and we didn't find one - we're done
+				return nil
+			}
+		} else {
+			if afterVal != nil && *afterVal == val {
+				// Asked to set a value and it's already set - we're done.
+				return nil
+			}
+		}
+
+		if afterVal == nil {
+			delete(md, key)
+		} else {
+			md[key] = *afterVal
+		}
+
+		// Attempt to write the new value now
+		op, err := config.clientCompute.Projects.SetCommonInstanceMetadata(
+			projectID,
+			&compute.Metadata{
+				Fingerprint: project.CommonInstanceMetadata.Fingerprint,
+				Items:       expandComputeMetadata(md),
+			},
+		).Do()
+
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[DEBUG] SetCommonInstanceMetadata: %d (%s)", op.Id, op.SelfLink)
+
+		return computeOperationWaitGlobal(config, op, project.Name, "SetCommonInstanceMetadata")
```

Comment:
This should be `computeOperationWait` as per https://github.com/terraform-providers/terraform-provider-google/pull/243